include/wil/resource.h

```diff
@@ -4954,6 +4956,16 @@ namespace wil
     typedef weak_any<shared_cotaskmem_string> weak_cotaskmem_string;
 #endif // __WIL_OBJBASE_H_STL
 
+#if defined(_SHLOBJ_H_) && !defined(__WIL_SHLOBJ_H__) && WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+#define __WIL_SHLOBJ_H__
+    typedef unique_any<LPITEMIDLIST, decltype(&::ILFree), ::ILFree> unique_itemidlist;

```

Comment:
LPITEMIDLIST is a problematic type. if you use [STRICT_TYPED_ITEMIDS](https://docs.microsoft.com/en-us/windows/desktop/api/shtypes/ns-shtypes-_itemidlist) it is re-defined as 3 distinct types, absolute, relative, child to create some type safety. all clients should use that to avoid common errors. Given that I'd like to see this support exclusive to that mode.